src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -239,6 +240,19 @@ class Direct$Type$Buffer$RW$$BO$
 #end[rw]
     }
 
+    // Constrain the capacity to int range throwing IAE if not possible
+    //
+    private static int clampCapacity(long capacity) {
+        try {
+            return Math.toIntExact(capacity);
+        } catch (ArithmeticException ignore) {
+            String msg = "JNI NewDirectByteBuffer passed illegal capacity: "
+                + capacity + (capacity < Integer.MIN_VALUE
+                ? " < Integer.MIN_VALUE" : " > Integer.MAX_VALUE");
```

Comment:
Why the check against `MIN_VALUE`?

Reply:
It was the lower bound for `toIntExact()` but not relevant any more.